Dockerfile

```diff
@@ -1,23 +1,42 @@
-FROM node:20-slim
-
+FROM node:20-slim as base
 RUN apt-get update && \
     apt-get install -y python3 curl git build-essential psmisc && \
     apt-get clean
 
 COPY . /app
 WORKDIR /app
 
-ENV DEV_CHAIN_ID=313371
-ENV PNPM_HOME="/pnpm"
-ENV PATH="$PNPM_HOME:$PATH"
-RUN corepack enable
 
 # Check when foundry supports dumping events
 # https://github.com/foundry-rs/foundry/issues/5906
 RUN curl -L https://foundry.paradigm.xyz | bash
-RUN ~/.foundry/bin/foundryup
 
+# install a previous versions that doesn't have the rpc method hardhat_metadata.
+# hardhat expects an integer but the output from anvil is an hex string
+RUN ~/.foundry/bin/foundryup -v nightly-34f684ddfacc5b2ed371353ba6f730c485616ffe
+
+ENV PATH="$PNPM_HOME:$PATH"
+
+RUN corepack enable
 RUN pnpm install
+
+FROM node:16-bullseye-slim as prod
```

Comment:
Was using v16 intentional (e.g. compatibility problems with newer images)?

Reply:
there was one dependency for hardware wallets in the past that reaquired v16 but we can try again to upgrade